bin/lib/fetchContent.js

```diff
@@ -1,18 +1,128 @@
-const individualUUIDs = require('./individualUUIDs');
 
 const CAPI_KEY = process.env.CAPI_KEY;
 if (! CAPI_KEY ) {
 	throw new Error('ERROR: CAPI_KEY not specified in env');
 }
 
 const CAPI_PATH = 'http://api.ft.com/enrichedcontent/';
+const SAPI_PATH = 'http://api.ft.com/content/search/v1';
+
+function constructSAPIQuery( params ) {
+
+	const defaults = {
+		queryString : "",
+	   maxResults : 1,
+		     offset : 0,
+		    aspects : [ "title"], // [ "title", "location", "summary", "lifecycle", "metadata"],
+		constraints : []
+	};
+
+	const combined = Object.assign({}, defaults, params);
+
+	let queryString = combined.queryString;
+	if (queryString == '' && combined.constraints.length > 0 ) {
+		queryString = combined.constraints.join(' and ');
+	}
+
+	const full = {
+  	"queryString": queryString,
+  	"queryContext" : {
+         "curations" : [ "ARTICLES", "BLOGS" ]
+		},
+  	"resultContext" : {
+			"maxResults" : `${combined.maxResults}`,
+		 	    "offset" : `${combined.offset}`,
+			   "aspects" : combined.aspects,
+			 "sortOrder" : "DESC",
+			 "sortField" : "lastPublishDateTime",
+			    // "facets" : {"names":["people"], "maxElements":-1}
+  	}
+	}
+
+	return full;
+}
+
+function search(params) {
+	const sapiUrl = `${SAPI_PATH}?apiKey=${CAPI_KEY}`;
+	const sapiQuery = constructSAPIQuery( params );
+	debug(`search: sapiQuery=${JSON.stringify(sapiQuery)}`);
+
+	return fetch(sapiUrl, {
+		 method: 'POST',
+       body: JSON.stringify(sapiQuery),
+		headers: {
+			'Content-Type' : 'application/json',
+		},
+	})
+	.then( res  => res.text() )
+	.then( text => {
+		debug(`search: res.text=${text}`);
+		return text;
+	})
+	.then( text => {
+		return {
+			params : params,
+			sapiObj : JSON.parse(text)
+		};
+	} )
+	;
+}
+
+function searchByUUID(uuid) {
+	return search({queryString: uuid});
+}
+
+function searchLastFewFirstFt(maxResults) {
+	return search({queryString: `brand:FirstFT`, maxResults: maxResults});
+}
+
+function extractFirstFtIds( sapiObj ){
+	let uuids = [];
+	if (! sapiObj.results ) {
+		debug(`extractFirstFtIds: no sapiObj.results`);
+	} else if (! sapiObj.results[0]) {
+		debug(`extractFirstFtIds: no sapiObj.results[0]`);
+	} else if (! sapiObj.results[0].results) {
+		debug(`extractFirstFtIds: no sapiObj.results[0].results`);
+	} else if (sapiObj.results[0].results.length == 0) {
+		debug(`extractFirstFtIds: sapiObj.results[0].results.length == 0`);
+	} else {
+		uuids = sapiObj.results[0].results.map( r => { return r.id; } );
+	}
+	return uuids
+}
+
+// <ft-content type=\"http://www.ft.com/ontology/content/Article\" url=\"http://api.ft.com/content/dd033082-49e9-11e7-a3f4-c742b9791d43\" title=\"www.ft.com\">paid up to $1bn</ft-content>
+const reFtContent = new RegExp(/<ft-content\s+type=\"http:\/\/www.ft.com\/ontology\/content\/Article\"\s+url=\"http:\/\/api.ft.com\/content\/([a-f0-9-]+)\"/, 'g');
+
+function getLastFewFirstFtMentionedUuids(maxResults, includeFirstFtUuids=false) {
```

Comment:
This function doesn't `catch` errors. If only one of the promises in `Promise.all` fails, the whole function will fail. Are errors being handled somewhere further up the chain?